include/lz4_stream.h

```diff
@@ -54,6 +54,26 @@ class basic_ostream : public std::ostream
   }
 
  private:
+  /**
+   * @brief RAII version of `LZ4F_compressionContext_t`.
+   */
+  struct CompressionContext
+  {
+    LZ4F_compressionContext_t ctx;
+
+    CompressionContext() {
+      size_t ret = LZ4F_createCompressionContext(&ctx, LZ4F_VERSION);
+      if (LZ4F_isError(ret) != 0) {
+        throw std::runtime_error(std::string("Failed to create LZ4 compression context: ")
+                                 + LZ4F_getErrorName(ret));
```

Comment:
Unrelated to this change, but reminds me that I have thought or creating a `lz4_category` that inherits from `std::error_category` and the throw `std::system_error` instead.

That way a more specific exception could be thrown as well as not having to add the LZ4 error string every time an exception object is constructed. Makes sense?

Reply:
Yes, I think specific exceptions are good so that they can be caught explicitly.

I'll leave that to you though :)